ui/src/app/job-list/table/table.component.html

```diff
@@ -67,7 +67,43 @@
     <ng-container matColumnDef="{{ df.field }}">
       <mat-header-cell *matHeaderCellDef>{{ df.display }}</mat-header-cell>
       <mat-cell class="additional-column" *matCellDef="let j">
-        <div class="display-field" *ngIf=!isStatusField(df)>{{ getFieldValue(j, df) }}</div>
+        <div class="display-field" *ngIf=canEdit(df)>
+          <div [ngSwitch]="getFieldType(df)">
+            <div *ngSwitchCase="'text'">
+              <div class="edit-field">
+                <span class="field-value">{{ getFieldValue(j, df) }}</span>
+                <button class="jobs-dropdown-arrow" mat-icon-button [matMenuTriggerFor]="menu" [class.visible]="showDropdownArrow(j)">
+                  <img src="{{ getDropdownArrowUrl() }}">
+                </button>
+              </div>
+              <mat-menu #menu="matMenu" class="wide-menu" xPosition="before" yPosition="below" [overlapTrigger]="false">
+                <mat-form-field>
+                  <input matInput
+                         #labelField
+                         value="{{ getFieldValue(j, df) }}"
+                         maxlength="255"
+                         (click)="$event.stopPropagation()">
+                  <mat-hint align="end">{{labelField.value?.length || 0}}/255</mat-hint>
+                </mat-form-field>
+                <button class="apply-link" (click)="setFieldValue(j, df.field, labelField.value)">Apply</button>
+              </mat-menu>
+            </div>
+            <mat-form-field *ngSwitchCase="'list'">
+              <mat-select #labelSelect
+                          [value]="getFieldValue(j, df) ? getFieldValue(j, df) : null"
+                          (change)="setFieldValue(j, df.field, labelSelect.value)">
+                <mat-option value="">(none)</mat-option>
+                <mat-option *ngFor="let fieldOption of getFieldOptions(df)"
+                            [value]="fieldOption"
+                            [attr.selected]="(getFieldValue(j, df) == fieldOption) ? 'selected' : null"
```

Comment:
nit: parens unneeded, ternary is generally very low precedence

Reply:
I don't feel too strongly on this, but this is a relatively common pattern and I don't recall seeing this style used before. Alternatively, extract this into a helper method, since ternaries as a whole are generally not considered very good for readability.

Reply:
removed the `()`